src/imgix-core-js.js

```diff
@@ -44,47 +38,25 @@
         this.settings[key] = val;
       }
 
-      if (Array.isArray(this.settings.domains)) { 
-        if (this.settings.domains.length > 1) {
-          console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
-        }
-        else if (this.settings.domains.length == 0) {
-          if (typeof(this.settings.domain) != "string" && this.settings.domain != null) {
-            throw new Error('ImgixClient.settings.domain only accepts a string argument');
-          }
-          else {
-            this.settings.domains = [this.settings.domain];
-          }
-        }
-      }
-      else {
-        this.settings.domains = [this.settings.domains];
+      if (typeof(this.settings.domain) != "string") {
```

Comment:
Does this work? This should not work. Should be:

```js
if (typeof this.settings.domain != "string") {
```

Reply:
Also, I'd suggest we just combine this with the second check, and just use "ImgixClient must be passed valid string domain" as the error message.

Reply:
I think so? It's what we did [last time](https://github.com/imgix/imgix-core-js/blob/ca889dc09dd153dfca889d7aaa6da2fde1cf7c5b/src/imgix-core-js.js#L52) but I can double check

Reply:
Oh god, I should have caught that. `typeof` is an operator, not a function. That needs to be fixed.